columnq/src/columnq.rs

```diff
@@ -23,31 +26,42 @@ pub struct ColumnQObjectStoreProvider {}
 impl ObjectStoreProvider for ColumnQObjectStoreProvider {
     fn get_by_url(&self, url: &Url) -> DatafusionResult<Arc<dyn object_store::ObjectStore>> {
         let url_schema = url.scheme();
-        match url_schema {
-            "s3" => {
-                let host = url.host_str().unwrap();
-                let mut s3_builder = AmazonS3Builder::from_env().with_bucket_name(host);
-                // for minio in CI
-                s3_builder = s3_builder.with_allow_http(true);
-
-                match s3_builder.build() {
-                    Ok(s3) => Ok(Arc::new(s3)),
-                    Err(err) => Err(DataFusionError::External(Box::new(err))),
-                }
-            },
-            "gs" => {
-                let host = url.host_str().unwrap();
-                let gcs_builder = GoogleCloudStorageBuilder::from_env().with_bucket_name(host);
-                match gcs_builder.build() {
-                    Ok(gcs) => Ok(Arc::new(gcs)),
-                    Err(err) => Err(DataFusionError::External(Box::new(err))),
-                }
-
-            },
-            _ => Err(DataFusionError::Execution(format!(
-                "Unsupported object store scheme {}",
-                url_schema
-            ))),
+        match BlobStoreType::try_from(url_schema) {
+            Err(err) => Err(DataFusionError::External(Box::new(err))),
+            Ok(blob_type) => match blob_type {
+                BlobStoreType::S3 => {
+                    let host = url.host_str().unwrap();
+                    let mut s3_builder = AmazonS3Builder::from_env().with_bucket_name(host);
+                    // for minio in CI
+                    s3_builder = s3_builder.with_allow_http(true);
+
+                    match s3_builder.build() {
+                        Ok(s3) => Ok(Arc::new(s3)),
+                        Err(err) => Err(DataFusionError::External(Box::new(err))),
+                    }
+                },
+                BlobStoreType::GCS => {
+                    let host = url.host_str().unwrap();
+                    let gcs_builder = GoogleCloudStorageBuilder::from_env().with_bucket_name(host);
+                    match gcs_builder.build() {
+                        Ok(gcs) => Ok(Arc::new(gcs)),
+                        Err(err) => Err(DataFusionError::External(Box::new(err))),
+                    }
+
+                },
+                BlobStoreType::Azure => {
+                    let host = url.host_str().unwrap();
```

Comment:
better avoid unwrap whenever possible, if there is no error to return, we should at least use `host_str().expect()`.

Reply:
thanks for the suggestion. TIL: https://jakedawkins.com/2020-04-16-unwrap-expect-rust/

I just handle the error properly in 94fc3ec